test/helpers/kubectl.go

```diff
@@ -1114,7 +1114,15 @@ func (kub *Kubectl) NamespaceLabel(namespace string, label string) *CmdRes {
 // the aforementioned desired state within timeout seconds. Returns false and
 // an error if the command failed or the timeout was exceeded.
 func (kub *Kubectl) WaitforPods(namespace string, filter string, timeout time.Duration) error {
-	return kub.waitForNPods(checkReady, namespace, filter, 0, timeout)
+	ginkgoext.By("WaitforPods(namespace=%q, filter=%q)", namespace, filter)
+	err := kub.waitForNPods(checkReady, namespace, filter, 0, timeout)
+	ginkgoext.By("WaitforPods(namespace=%q, filter=%q) => %v", namespace, filter, err)
+	if err != nil {
+		desc := kub.ExecShort(fmt.Sprintf("%s describe pods -n %s %s", KubectlCmd, namespace, filter))
+		ginkgoext.By("kubectl describe pods -n %s %s:\n=== STDOUT ===\n%s\n=== STDERR ===\n%s\n===\n",
```

Comment:
Unless you are attached to formatting, please consider using `desc.GetDebugMessage`